pkg/kubelet/kubelet_pods.go

```diff
@@ -2072,6 +2072,17 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *v1.Pod, podStatus *kubecon
 		if oldStatus.Resources == nil {
 			oldStatus.Resources = &v1.ResourceRequirements{}
 		}
+
+		convertCustomResources := func(inResources, outResources v1.ResourceList) {
+			for resourceName, resourceQuantity := range inResources {
+				if resourceName == v1.ResourceCPU || resourceName == v1.ResourceMemory ||
+					resourceName == v1.ResourceStorage || resourceName == v1.ResourceEphemeralStorage {
```

Comment:
why `resourceName == v1.ResourceStorage` excluded here and have no special processing below? Is it intended? Please also add a test on it

Reply:
I'm not sure I understand the question.
When `resourceName == v1.ResourceStorage` the resource is not a custom resource, and it's already being handled below:
https://github.com/kubernetes/kubernetes/blob/a31030543c47aac36cf323b885cfb6d8b0a2435f/pkg/kubelet/kubelet_pods.go#L2095-L2096
https://github.com/kubernetes/kubernetes/blob/a31030543c47aac36cf323b885cfb6d8b0a2435f/pkg/kubelet/kubelet_pods.go#L2112-L2113

What am I missing?

Reply:
I see special handling for ephemeral storage, but not `v1.ResourceStorage`. Moreover, the ephemeral storage conversion is doing the same thing that's being done here, so it might as well be combined with this conversion.

Reply:
Thanks for the clarifications! :)
Done! Also added a unit test.